tests/test_mesh.py

```diff
@@ -670,6 +671,35 @@ def test_nativetag():
     # deleting tag
     del m.f[:]
 
+def test_del_nativetag():
+    mats = {
+        0: Material({'H1': 1.0, 'K39': 1.0}, density=42.0),
+        1: Material({'H1': 0.1, 'O16': 1.0}, density=43.0),
+        2: Material({'He4': 42.0}, density=44.0),
+        3: Material({'Tm171': 171.0}, density=45.0),
+        }
+    m = gen_mesh(mats=mats)
+    m.f = NativeMeshTag(mesh=m, name='f')
+    m.f[:] = [1.0, 2.0, 3.0, 4.0]
+
+    assert_raises(ValueError, m.delete_tag,-12)
+
+    import sys
+
+    # make a new reference to the tag that can will not
+    # be deleted
+    tag_ref = m.f
+
+    # deleting tag
+    m.delete_tag('f')
+
+    # ensure that there are only 2 references to this tag
+    # 1. is the tag_ref created above
+    # 2. is the one that automatically is the temporary
+    #    reference created as the argument to getrefcount
+    assert_equal(2,sys.getrefcount(tag_ref))
```

Comment:
We should probably check the PyMOAB instance itself to make sure that it's been deleted at that level too.

Reply:
If that's what we want...

Reply:
If I call `tag_get_handle()` with `create_if_missing = False` will it return `None` or raise an exception?

Reply:
Looks like it currently does NOT remove the PyMOAB tag... stay tuned

Reply:
Missed that question about `tag_get_handle`. It will raise an exception if it cannot find the tag.